R/utilities.R

```diff
@@ -186,19 +190,24 @@ getPgParams = function(path = '~/.pgpass') {
 
 
 writeTableInChunks = function(path, con, nRowsPerChunk, overwrite, tableName) {
+  # get total number of lines
+  n = R.utils::countLines(path)
+
+  # read first row to get data types, but don't send it to database
+  d = data.table::fread(path, nrows = 1L)
+
   # will be inefficient if file is compressed
-  if (DBI::dbExistsTable(con, tableName)) {
+  if (!(DBI::dbExistsTable(con, tableName) && !isTRUE(overwrite))) {
```

Comment:
I don't understand the reasoning behind the complicated logic of the changed code. Why should the function not remove the old table, if it exists and overwrite is true?